patches/tModLoader/Terraria/ModLoader/UI/UIModSourceItem.cs

```diff
@@ -383,4 +335,95 @@ internal static void PublishModCommandLine(string modName)
 		Steamworks.SteamAPI.Shutdown();
 		Environment.Exit(0);
 	}
+
+	private void StartUpgradeTasks()
+	{
+		langFileTask = Task.Run(
+			() => Directory.GetFiles(_mod, "*.lang", SearchOption.AllDirectories),
+			_modSourcesCts.Token
+		);
+
+		sourceUpgradeTask = Task.Run(
+			() => SourceManagement.SourceUpgradeNeeded(_mod),
+			_modSourcesCts.Token
+		);
```

Comment:
While we're at it, it would be good to exclude common paths that aren't relevant (`.git/`, `.vs/`, `obj/`, `bin/`). A good option would be ignoring directories that fall under buildIgnore, but I'm not sure whether that's accessible in this context let alone whether it's reasonable to implement.
Either way, the aforementioned directories are reliably ignorable.